src/drivers/uart/uart.c

```diff
@@ -83,13 +97,57 @@ w_status_t uart_init(uart_channel_t channel, UART_HandleTypeDef *huart, uint32_t
         return W_FAILURE;
     }
 
+    // Register the transmit-complete ISR for this UART channel
+    hal_status =
+        HAL_UART_RegisterCallback(huart, HAL_UART_TX_COMPLETE_CB_ID, HAL_UART_TxCpltCallback);
+    if (hal_status != HAL_OK) {
+        return W_FAILURE; // error occured in HAL_UART_RegisterCallback function
+    }
+
     // Start first reception
     if (HAL_UARTEx_ReceiveToIdle_IT(huart, handle->rx_msgs[0].data, UART_MAX_LEN) != HAL_OK) {
         return W_IO_ERROR;
     }
 
     return W_SUCCESS;
 }
+/**
+ * @brief Write to the specified UART channel
+ * @details // One task can write to a channel at once, and concurrent calls will block for 'timeout
+ * @param channel UART channel to write to
+ * @param data Buffer to store the data to send
+ * @param length Pointer to store the length of the sending message
+ * @param timeout_ms Timeout in milliseconds
+ * @return Status of the write operation
+ */
+w_status_t
+uart_write(uart_channel_t channel, uint8_t *buffer, uint8_t *length, uint32_t timeout_ms) {
```

Comment:
I'm a little confused about the type of `length` here: (1) if `UART_MAX_LEN` is 256 it won't fit in 8 bits, (2) the design doc says `length` would be `uint16_t *`, also (3) the corresponding parameter to `HAL_UART_Transmit_IT` is `uint16_t`.

On that note, is there a reason why `uart_msg_t`'s `len` field is 32-bit if these functions and the HAL functions only accept a 16-bit size?